packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-settings/pipeline-extra-settings.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import type { ConnectedProps } from 'react-redux';
+import {
+  Icon,
+  Toggle,
+  Label,
+  css,
+  spacing,
+  IconButton,
+} from '@mongodb-js/compass-components';
+import { toggleSettingsIsExpanded } from '../../../modules/settings';
+import { toggleAutoPreview } from '../../../modules/auto-preview';
+
+const extraSettingsGroupStyles = css({
+  display: 'flex',
+});
+
+const buttonStyles = css({
+  marginLeft: spacing[1],
+  marginRight: spacing[1],
+});
+
+const toggleStyles = css({
+  display: 'flex',
+  alignItems: 'flex-end',
+  marginLeft: spacing[1],
+  marginRight: spacing[1],
+});
+
+const toggleLabelStyles = css({
+  marginBottom: 0,
+  textTransform: 'uppercase',
+  marginLeft: spacing[1],
+});
+
+const PipelineExtraSettings: React.FunctionComponent<PipelineExtraSettingsProps> =
+  ({ onToggleAutoPreview, onToggleSettings }) => {
+    return (
+      <div className={extraSettingsGroupStyles}>
+        <div className={toggleStyles}>
+          <Toggle
+            id="auto-preview"
+            size="small"
+            aria-label="Toggle Auto Preview"
+            onChange={() => onToggleAutoPreview()}
+          />
+          <Label className={toggleLabelStyles} htmlFor="auto-preview">
+            Auto Preview
+          </Label>
```

Comment:
Seems like this needs some style adjustments due to bootstrap styles

<img width="189" alt="image" src="https://user-images.githubusercontent.com/5036933/158768439-4ecc34d0-6c73-4e23-bc8c-c659ea0512df.png">